### PR TITLE
PR: Fixes (GEOS-7672) Layer publishing regression: need to select the…

### PR DESCRIPTION
**Pull request by afabiani**
… store and Published flag is false

https://osgeo-org.atlassian.net/browse/GEOS-7672

### CODE REVIEW

File: src/main/src/main/java/org/geoserver/catalog/ResourcePool.java [hunk 1]
```diff
@@ -2343,35 +2345,38 @@ private void flushState(ContentDataStore contentDataStore, String nativeName)
...
     public DataStoreInfo clone(final DataStoreInfo source, boolean allowEnvParametrization) {
-        DataStoreInfo target = catalog.getFactory().createDataStore();
-        target.setEnabled(source.isEnabled());
-        target.setName(source.getName());
-        target.setDescription(source.getDescription());
-        target.setWorkspace(source.getWorkspace());
-        target.setType(source.getType());
+        DataStoreInfo target;
+        try {
+            target = (DataStoreInfo) SerializationUtils.clone(source);
```
Comment (jodygarnett) [inline src/main/src/main/java/org/geoserver/catalog/ResourcePool.java]: I guess this is to get out of breaking the code each time a new field is added?

I am used to using copy constructors to hide individual fields being set ...

File: src/web/core/src/main/java/org/geoserver/web/data/store/CoverageStoreNewPage.java [hunk 2]
```diff
@@ -82,7 +82,8 @@ protected void onSuccessfulSave(final CoverageStoreInfo info, final Catalog cata
         CoverageStoreInfo expandedStore = getCatalog().getFactory().createCoverageStore();
         try {
             clone(savedStore, expandedStore);
-            layerChooserPage = new NewLayerPage(expandedStore.getId());
+            // The ID is assigned by the catalog and therefore cannot be cloned
+            layerChooserPage = new NewLayerPage(savedStore.getId());
```
Comment (jodygarnett) [inline src/web/core/src/main/java/org/geoserver/web/data/store/CoverageStoreNewPage.java]: This appears to be the only functional change

**Comment by jodygarnett** [comment]
@afabiani this is a fix right, are we backporting to 2.9.x, 2.8.x

**Comment by jodygarnett** [comment]
Merge when ready!

File: src/main/src/main/java/org/geoserver/catalog/ResourcePool.java (hunk 1 above)
Comment (afabiani) [inline src/main/src/main/java/org/geoserver/catalog/ResourcePool.java]: Yep

File: src/web/core/src/main/java/org/geoserver/web/data/store/CoverageStoreNewPage.java (hunk 2 above)
Comment (afabiani) [inline src/web/core/src/main/java/org/geoserver/web/data/store/CoverageStoreNewPage.java]: Yep, also only the savedStore has the ID setted at this time.

**Pull request merged**